test/e2e/catalog_e2e_test.go

```diff
@@ -1056,27 +1057,30 @@ var _ = Describe("Catalog represents a store of bundles which OLM can use to ins
 		}
 
 		By("creating a catalog source")
-		source, err := crc.OperatorsV1alpha1().CatalogSources(source.GetNamespace()).Create(context.TODO(), source, metav1.CreateOptions{})
-		Expect(err).ToNot(HaveOccurred())
-
-		By("update the catalog source with template annotation")
 
-		source, err = crc.OperatorsV1alpha1().CatalogSources(source.GetNamespace()).Get(context.TODO(), source.GetName(), metav1.GetOptions{})
-		Expect(err).ShouldNot(HaveOccurred(), "error getting catalog source")
-
-		// create an annotation using the kube templates
+		var err error
+		Eventually(func() error {
+			source, err = crc.OperatorsV1alpha1().CatalogSources(source.GetNamespace()).Create(context.TODO(), source, metav1.CreateOptions{})
+			return err
+		}).Should(Succeed())
 
-		source.SetAnnotations(map[string]string{catalogsource.CatalogImageTemplateAnnotation: fmt.Sprintf("quay.io/olmtest/catsrc-update-test:%s.%s.%s", catalogsource.TemplKubeMajorV, catalogsource.TemplKubeMinorV, catalogsource.TemplKubePatchV)})
+		By("update the catalog source with template annotation")
 
 		// Update the catalog image
 		Eventually(func() error {
+			source, err = crc.OperatorsV1alpha1().CatalogSources(source.GetNamespace()).Get(context.TODO(), source.GetName(), metav1.GetOptions{})
+			Expect(err).ShouldNot(HaveOccurred(), "error getting catalog source")
```

Comment:
nit: should this error be handled in the typical 
```
if err != nil {
  return err
}
```
instead of an `Expect()` since we are potentially expecting retries inside of an `Eventually()`